HealthcareApp/src/HealthcareApi/Controllers/MetricsController.cs

```diff
@@ -0,0 +1,53 @@
+using Healthcare.Application.Metrics.Queries.GetDoctorAppointmentsCount;
+using Healthcare.Application.Metrics.Queries.GetDoctorsWithAppointments;
+using Healthcare.Application.Metrics.Queries.GetMedianOfDurationTime;
+using Healthcare.Application.Metrics.Queries.GetUsersCountByRole;
+using Healthcare.Application.Metrics.Queries.GetVarianceOfDurationTime;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace HealthcareApi.Controllers;
+
+[Route("api/[controller]")]
+public class MetricsController : Controller
+{
+    private readonly IMediator _mediator;
```

Comment:
I liked that you use MediatR. Honestly, I have never used it before, but because of you, I will start.

Reply:
about the comparison, i wrote to Sasha, that I will write it a bit later. Regarding the MediatR, thank you, that is nice library, I recommend to use it